app-next/components/MessageUpdater.jsx

```diff
@@ -0,0 +1,28 @@
+"use client";
+import { useEffect } from "react";
+import { getMessages } from "@/roomActions";
+
+const MessageUpdater = ({ roomId, onUpdate }) => {
+  useEffect(() => {
+    const fetchMessages = async () => {
+      try {
+        const newMessages = await getMessages(roomId);
```

Comment:
I would put a limit on the number of messages, or maybe pass a timestamp so only messages after the timestamp are passed. And you can ask for messages since the timestamp of the last message in local state, so this way you only receive new messages 😄 

The reason I'm saying this is as the number of messages increases, the time it will take to poll will also increase, and you might start to see weird behaviour once it exceeds 1s.

You can also decrease the polling to about 3s without noticing much difference.